B/BOLT/build_tarballs.jl

```diff
@@ -0,0 +1,179 @@
+CMAKE_FLAGS+=(-DCMAKE_INSTALL_PREFIX=${prefix})
+CMAKE_FLAGS+=(-DCMAKE_CROSSCOMPILING=True)
+
+# Tell LLVM where our pre-built tblgen tools are
+CMAKE_FLAGS+=(-DLLVM_TABLEGEN=${WORKSPACE}/bootstrap/bin/llvm-tblgen)
+CMAKE_FLAGS+=(-DLLVM_CONFIG_PATH=${WORKSPACE}/bootstrap/bin/llvm-config)
+
+# Explicitly use our cmake toolchain file
+CMAKE_FLAGS+=(-DCMAKE_TOOLCHAIN_FILE=${CMAKE_TARGET_TOOLCHAIN})
+
+# Manually set the host triplet, as otherwise on some platforms it tries to guess using
+# `ld -v`, which is hilariously wrong.
+CMAKE_FLAGS+=(-DLLVM_HOST_TRIPLE=${target})
+
+# Most targets use the actual target string, but we disagree on `aarch64-darwin` and `arm64-darwin`
+CMAKE_TARGET=${target}
+
+if [[ "${target}" == *apple* ]]; then
+    # On OSX, we need to override LLVM's looking around for our SDK
+    CMAKE_FLAGS+=(-DDARWIN_macosx_CACHED_SYSROOT:STRING=/opt/${target}/${target}/sys-root)
+    CMAKE_FLAGS+=(-DDARWIN_macosx_OVERRIDE_SDK_VERSION:STRING="${MACOSX_DEPLOYMENT_TARGET}")
+    # We need to link against libc++ on OSX
+    CMAKE_FLAGS+=(-DLLVM_ENABLE_LIBCXX=ON)
+
+    # If we're building for Apple, CMake gets confused with `aarch64-apple-darwin` and instead prefers
+    # `arm64-apple-darwin`.  If this issue persists, we may have to change our triplet printing.
+    if [[ "${target}" == aarch64* ]]; then
+        CMAKE_TARGET=arm64-${target#*-}
+    fi
+
+    if [[ "${target}" == x86_64* ]]; then
+        CMAKE_FLAGS+=(-DDARWIN_osx_BUILTIN_ARCHS="x86_64")
+        CMAKE_FLAGS+=(-DDARWIN_osx_ARCHS="x86_64")
+    fi
+fi
+
+GCC_VERSION=$(gcc --version | head -1 | awk '{ print $3 }' | cut -d. -f1)
+if [[ $version -le 10 && "${target}" == aarch64-linux* ]]; then
+    CMAKE_C_FLAGS+=(-mno-outline-atomics)
+    CMAKE_CPP_FLAGS+=(-mno-outline-atomics)
+fi
+
+# Tell LLVM which compiler target to use, because it loses track for some reason
+CMAKE_FLAGS+=(-DCMAKE_C_COMPILER_TARGET=${CMAKE_TARGET})
+CMAKE_FLAGS+=(-DCMAKE_CXX_COMPILER_TARGET=${CMAKE_TARGET})
+CMAKE_FLAGS+=(-DCMAKE_ASM_COMPILER_TARGET=${CMAKE_TARGET})
+
+# Set the bug report URL to the Julia issue tracker
+CMAKE_FLAGS+=(-DBUG_REPORT_URL="https://github.com/julialang/julia")
+
+cmake -GNinja ${LLVM_SRCDIR} ${CMAKE_FLAGS[@]} -DCMAKE_CXX_FLAGS=\"${CMAKE_CPP_FLAGS[*]} ${CMAKE_CXX_FLAGS[*]}\" -DCMAKE_C_FLAGS=\"${CMAKE_C_FLAGS[*]}\"
+ninja -j${nproc} -vv
+
+# Install!
+ninja install
+
+install_license ${WORKSPACE}/srcdir/llvm-project/llvm/LICENSE.TXT
+"""
+
+function configure_build(ARGS, version; git_path="https://github.com/llvm/llvm-project.git",
+    git_ver=llvm_tags[version])
+
+    sources = BinaryBuilder.AbstractSource[GitSource(git_path, git_ver)]
+    if version == v"16"
+        push!(sources, DirectorySource("./bundled"))
+    end
+
+    platforms = expand_cxxstring_abis(supported_platforms())
+    filter!(p -> arch(p) ∈ ("x86_64", "aarch64") && os(p) ∈ ("linux", "macos"), platforms)
+
+    products = [ExecutableProduct("llvm-bolt", :llvm_bolt, "bin")]
```

Comment:
`ExecutableProduct` by default searches files in `bin/`
```suggestion
    products = [ExecutableProduct("llvm-bolt", :llvm_bolt)]
```